### PR TITLE
Feature/containerize application

### PR DESCRIPTION
**Pull request by mtheoryx**
This was created "while stream" not "on-stream" but I wanted to pitch in and do something relatively straightforward.

I updated the readme with very basic info. I am happy to add more if we start getting a lot of traction on the repo and requests.

@noopkat I stuck with node 8.14, happy to bump that to 10 something if you'd like.

### CODE REVIEW

**Comment by whitfin** [comment]
@mtheoryx should the `package-lock.json` also be copied into the Dockerfile before the install? I've often wondered this on personal projects and I'm curious what the best practice for this is!

**Comment by mtheoryx** [comment]
@whitfin IMHO, I would say no. Let the container environment dictate what gets pulled own. For example, anything relying on something like fsevents will differ between environments. So you don't want that cooked on a mac, and moved into alpine linux. The goal is to really have a solid image that gets baked and pushed to something like docker hub.

**Comment by mtheoryx** [comment]
Closes #7

**Comment by noopkat** [comment]
@mtheoryx this is fantastic thanks!

For my own FYI, could you please explain the strategy behind creating those volumes? As opposed to keeping everything inside the image's file system? I assume this is the flow for being able to develop it locally, but I worry this would break things if the container was built while running on anything other than a Unix machine.

Thanks again!

**Comment by mtheoryx** [comment]
@noopkat No problem!

So, using the Docker Compose method to bring up the container is really for development. The volume mapping lets you edit the files on your host file system and see them reflected in a running container.

However, if you want to build an image and publish it for others to pull down and run off of Docker Hub (or other registry) there is no volume mapping, and the files are pulled in and baked into the image, as-is.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  app:
+    build: .
+    image: mxchip-gifs
+    container_name: mxchip-gifs
+    command: node server.js
+    ports:
+      - 3000:3000
+    volumes:
+      - ./:/usr/app
+      - /usr/app/node_modules
```
Comment (noopkat) [inline docker-compose.yml:13]: I think I'm totally missing something, but where does this volume map to on a non unix system?

**Comment by noopkat** [comment]
@mtheoryx thanks for the explanation 🙇‍♀️  I left a further question on the compose file!

**Comment by sergiofteixeira** [comment]
`./:/usr/app`
It maps ur current directory to /usr/app on the docker machine , like a symlink, if u edit on ur current directory it will affect the docker machine.
ur machine directory:docker machine dir

the other volume just copies ur node modules to the machine everytime u do a docker compose up.

**Comment by noopkat** [comment]
@Swashed thanks for explaining. I understood the first mapping (I know about the colon structure). I wasn't familiar with omitting the colon in the second mapping and what that format signified.

**Comment by noopkat** [comment]
@mtheoryx I'm going to push up a commit to this branch that both resolves the merge conflicts that I caused and also moving your instructions to a contributing guide instead of the README. I hope that's ok with you!

File: docker-compose.yml (hunk 1 above)
Comment (mtheoryx) [reply docker-compose.yml:13]: For official reference, there are 3 types of volume mappings. On line 12, we are using the first example.  A `host` volume.

In line 13 we are using an `anonymous` volume. This really just means that we don't care where Docker Engine chooses to store it on our machine (and the location will depend on what system you're running Docker on Mac/Windows/Linux).

We aren't using a `named` volume anywhere here, but that's the third example.

https://success.docker.com/article/different-types-of-volumes

**Comment by mtheoryx** [comment]
Sounds great to me @noopkat !

**Comment by noopkat** [comment]
I rebased and force pushed, I hope that's ok 🙈

**Pull request merged**